packages/objectloader2/src/operations/downloader.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { beforeEach, describe, expect, test } from 'vitest'
+import createFetchMock from 'vitest-fetch-mock'
+import { vi } from 'vitest'
+import AsyncGeneratorQueue from '../helpers/asyncGeneratorQueue.js'
+import { Item } from '../types/types.js'
+import Downloader from './downloader.js'
+import CacheDatabase from './database.js'
+
+const fetchMocker = createFetchMock(vi)
+
+// sets globalThis.fetch and globalThis.fetchMock to our mocked version
+fetchMocker.enableMocks()
+
+describe('downloader', () => {
+  beforeEach(() => {
+    fetchMocker.resetMocks()
+  })
+  test('download batch', async () => {
```

Comment:
its good practice to make sure the final test titles are human readable sentences. the resulting string ends up being a combination of ancestor describe titles and the test's title.

so if you have a describe('downloader') and a test('downloads batch successfully'), that ends up building a nice human readable description for the test: downloader downloads batch successfully